basket/main.py

```diff
@@ -46,12 +71,26 @@ class Basket(object):
     # gets out of control.
     os = os
     tarfile = tarfile
-    urllib = urllib
+    urllib = urllib2
     zipfile = zipfile
     err = sys.stderr
     out = sys.stdout
     root = os.environ.get('BASKET_ROOT') or os.path.expanduser('~/.basket')
 
+    # Support http_proxy environment
+    # because of usage xmlrpclib and urllib we need to support both mechanisms
+    if os.environ.get('http_proxy'):
+        # Extract http_proxy from environment
+        proxy = ''.join(re.findall(r'[\d\.]+:?', os.environ.get('http_proxy')))
+
+        # Setup proxy for both: xmlrpclib and urllib2
+        transport = ProxiedTransport(proxy)
+        proxy_support = urllib2.ProxyHandler({'https': proxy })
+        opener = urllib2.build_opener(proxy_support, urllib2.HTTPHandler)
+        urllib2.install_opener(opener)
+    else:
+        transport = None
```

Comment:
That's a lot of logic at the class-level and it sets class-level variables that are not used. Perhaps you could move the code to the class constructor (`__init__`)?